### ISSUE TITLE
Incorrect type of cursor value

### ISSUE THREAD

**Issue by vitonsky** [jakearchibald/idb#286]
I have an IDB scheme https://github.com/translate-tools/linguist/blob/4da90de9e406442ab78df591bad04d7e6b93cdf3/src/requests/backend/translations/idb/schema/v2.ts#L8-L16

Here i use cursor value https://github.com/translate-tools/linguist/blob/4da90de9e406442ab78df591bad04d7e6b93cdf3/src/requests/backend/translations/data.ts#L135

And i found that type of cursor value incorrect, when my test failed.
The type is not consider a "keyPath" that i defined here https://github.com/translate-tools/linguist/blob/4da90de9e406442ab78df591bad04d7e6b93cdf3/src/requests/backend/translations/idb/schema/v2.ts#L59-L62

So cursor value contains not only properties of type `ITranslationEntry` but also property `id` that is auto incremented number.

It's my temporary fix for test https://github.com/translate-tools/linguist/blob/4da90de9e406442ab78df591bad04d7e6b93cdf3/src/requests/backend/translations/data.test.ts#L40-L42 and i have to refactor a lot of code to remove this unnecessary property.

Let's improve types. We can add a `keyPath` to a IDB scheme and merge it to a cursor value type and in another places when we get values from IDB

**Comment by jakearchibald** [comment]
To make it easier for me to debug this, can you reproduce the issue in the Typescript playground? [Here's a demo](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzhMBTAdgEQEIBo44DKAxgBaogCGcAvnAGZQQhwDkwAJgEasDcAUP2DoYqKPUrFUcALIBPHHFQAPUeg4BnAthLkqifnDhgmHAK7EYGgFwGjRgG6UANmdS2Eh+0fSUQ7uA0YKGEAcwFvIxNgKVt0MxAuMQjIkwhzSwBhdICgkPRwryMaFKMAa1Q5Wzyw0rhhDhVUG0Q2LjkAWmipVjiEpKhaFJL+GkFKDTl0YgYzaZhgCHQ4RpAIAAoASjs4YiWgla44AF44SgB3SmB4FAwcAB55HAA+ddYQTu5WfABGfE9vGYwKEoJRGutuNsAZE9ugDmkMjBCDBoNJTtwAHTEKCoSiiADyXAAVqhLMjUW8ERYrN8dpE4BU5AAFPGkWysKlZHLfIr2GibOpRUzU8k4rE4vGoACS6hUb3aXRCPXwHKVqFYAt5NFwXn5Ai8sIOMGUJ0OGOClDhkgWS0pwssGg1KUN8GIZigGmgpouV3gxoxQVRGNu6Ey7s9UC2KWA9Dg6zdHugUN5LrgTlcaN24egGPTbjqhogzlQGOcEFC6zzqE1xVGQA) that shows a cursor with a correct type.

**Comment by vitonsky** [comment]
@jakearchibald yes, [demo with broken types](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzhMBTAdgEQEIBo44DKAxgBaogCGcAvnAGZQQhwDkwAJgEasDcAUP2DoYqKPUrFUcALIBPHHFQAPUeg4BnAthLkqifnDhgmHAK7EYGgFwGjRgG6UANmdS2Eh+0fSUQ7uA0YKGEAcwFvIxNgKVt0MxAuMQj7GhSjAGtUOVsgkPRwryNhDhVUG0Q2LjkAWmipVjiEpKhaFLT+GkFKDTl0YgYzfphgCHQ4UpAIAAoASjs4YjGgia44AF44SgB3SmB4FAwcAB55HAA+adYQWu5WfABGfE9vMzBQqEpS6e55l8ilugViYIOZLIQYNBpJtuAA6YhQVCUUQAeS4ACtUODIYiriCwVZ7gtInAsnIAArI0i2Vj4iwwADCoNQ9yKqVm6XsdOxUPhiORqAAkuoVFdqnUQg18LTJSyOWyaLgvDR5V5ASsYMoNqtYcFKEDJCMxnjTPSNKx5UZ1fBiGYoBpoNqdnt4JrYUFeYd0Ay7Q6oHMUsB6HBprb7dA-mzrXAnK5oYtfdBYbG3Jz1RBnKhYc4IKFpinUJbaJ0gA)

**Comment by vitonsky** [comment]
We have to improve type for `createObjectStore`, to require exists property for keyPath.
We must consider case with a composite keyPath

**Comment by jakearchibald** [comment]
It looks like you've just missed part of the value definition? [This works](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzhMBTAdgEQEIBo44DKAxgBaogCGcAvnAGZQQhwDkwAJgEasDcAUP2DoYqKPUrFUcALIBPHHFQAPUeg4BnAthLkqifnDhgmHAK7EYGgFwGjRgG6UANmdS2Eh+0fSUQ7uA0YKGEAcwFvIxNgKVt0MxAuMQjIkwhzSwBhdICgkPRwryMaFKMAa1Q5Wzyw0rhhDhVUG0Q2LjkAWmipVjiEpKhaFJL+GkFKDTl0YgYzaZhgCHQ4RpAIAAoASjs4YiWgla44AF44SgB3SmB4FAwcAB55HAA+ddYQTu5WfABGfE9vGYwKEoJRGutuNsAZE9ugDmkMjBCDBoNJTtwAHTEKCoSiiADyXAAVqhLMjUW8ERYrN8dpE4BU5AAFPGkWysKlZHLfIr2GibOpRUzU8k4rE4vGoACS6hUb3aXRCPXwHKVqFYAt5NFwXn5Ai8sIOMGUJ0OGOClDhkgWS0pwssGg1KUN8GIZigGmgpouV3gxoxQVRGNu6Ey7s9UC2KWA9Dg6zdHugUN5LrgTlcaN24egGPTbjqhogzlQGOcEFC6zzqE1xVGQA).

**Comment by vitonsky** [comment]
@jakearchibald you're right, we use typescript exactly to avoid data misses. Types for `createObjectStore` must make impossible a missed property and generate compile time error for such cases

**Comment by jakearchibald** [comment]
I'll accept a PR for this if it works, but `keyPath` is pretty complex.

**Comment by vitonsky** [comment]
@jakearchibald agree about complexity, maybe i will try to fix when will have time. Let's mark this as bug and needs help

**Comment by dumbmatter** [comment]
I think my PR #252 solves this issue - add and put accept an object with or without the auto incrementing keyPath being present, but any object returned from the database has the keyPath value defined. [Example](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzhMBTAdgEQEIBo44DKAxgBaogCGcAvnAGZQQhwDkAAgCYCuIARlRgxUUAPTBOfVgG4AULODphUepWKo4AWQCeOOKgAew9JwDOBbCXJVEsuHDBMexGKYBct+-YBulADbcqB4Idl726JQgQXCmMFCKAOZyYfaOwOoe6Lx8IskpjhDOMADChdFZ-Lmh9jR59gDWqNqZ2VVhipyGqO6IbHzaALRp6qwtlVC0dXCU3DAQAJLoxFAUGDAA0k0ACpQwpB6sBUWlnay4obWyNPKUptpLDNxLMMAQ6HCdIBAAFACUnnBiG9Yh8+HAALzTADulGA8BQGBwAB4dDgAHzfVggQaSM5wACM+BCYW4YASUEonW+kn+xJSQPQIKO3BchDmKwhoIAdMtULtUAB5PgAK1QrPZqExzJcpjxdJScEa2h2ewO0pKZTO1RSMzmi15USUHjigXOCpovymqScLJgbOgqB5K35i06Bkx-SG8RG+EO3tQrEt2poZtoQdCDJBMAMnMkXLilEZahebylNpleNYzs4UPiwkDeUj8GI3CgpmgnMoMLhcGjXNiDq5CPQxVL5agfzywHocG+JbL0Fp2qLcF8AQ0kP77a5Y8CU0jED8jr8EAS31nqCDNXk9lEojg2FmtdIwHMsWAfj8cCh0HqpgAhKEq7D4HWGysuZTON95XAIlEDgAMQgCAtTCYZogAVlDC05BoIA)

**Comment by jakearchibald** [comment]
I'll find some time to take a close look at that. Sorry I haven't so far.
